autogluon/task/tabular_prediction/predictor.py

```diff
@@ -421,11 +421,10 @@ def fit_summary(self, verbosity=3):
                 print("*** Details of Hyperparameter optimization ***")
                 for model_type in hpo_results:
                     hpo_model = hpo_results[model_type]
-                    print("HPO for %s model:  Num. configurations tried = %s, Time spent = %s, Search strategy = %s"
-                          % (model_type, len(hpo_model['trial_info']), hpo_model['total_time'], hpo_model['search_strategy']))
-                    print("Best hyperparameter-configuration (validation-performance: %s = %s):"
-                          % (self.eval_metric, hpo_model['validation_performance']))
-                    print(hpo_model['best_config'])
+                    if 'trial_info' in hpo_model:
```

Comment:
We should probably just wrap the first print statement in the `if` statement

Reply:
sorry can you clarify what you mean?

Reply:
The `trial_info` key is only used in the first `print` statement, so you don't need to protect the other two print statements with this check

Reply:
Oh actually those other print statements implicitly still depend on the first one to make sense, so that's why they're in there as well